File: cpp/day_02b.cpp
#include <algorithm>
#include <fstream>
#include <iostream>
#include <unordered_set>

int main() {
  std::ifstream file;
  file.open("../input/day_02_input");
  std::unordered_set<int> entries;
  char dash, letter, colon;
  std::string pwd;
  int i1, i2;
  int count = 0;
  while (file >> i1 >> dash >> i2 >> letter >> colon >> pwd) {
    const bool valid = (pwd[i1 - 1] == letter) ^ (pwd[i2 - 1] == letter);
    if (valid) {
      ++count;
    }
  }
  std::cout << count << '\n';
  return count;
}

File: 2020/python/day_02b.py
#!/usr/bin/env python3

import re


def main():
    f = open("../input/day_02_input")
    pattern = re.compile(r"(\d+)-(\d+) ([a-z]): (.*)\n")
    count = 0
    for line in f:
        low, high, letter, pwd = re.match(pattern, line).groups()
        if (pwd[int(low) - 1] == letter) ^ (pwd[int(high) - 1] == letter):
            count += 1
    print(count)
    return count


if __name__ == "__main__":
    main()
